cmd/agent/app/integrations.go

```diff
@@ -359,19 +436,33 @@ func getIntegrationVersion(integration string, cachePath string) (string, error)
 			errMsg = err.Error()
 		}
 
-		return "", fmt.Errorf("error executing pip freeze: %s", errMsg)
+		return nil, fmt.Errorf("error executing pip freeze: %s", errMsg)
+	}
+
+	version, err := getVersionFromReqLine(integration, string(output))
+	if err != nil {
+		return nil, err
 	}
 
-	exp, err := regexp.Compile(fmt.Sprintf(pipFreezeOutputPattern, integration))
+	return version, nil
+}
+
+func getVersionFromReqLine(integration string, lines string) (*integrationVersion, error) {
+	exp, err := regexp.Compile(fmt.Sprintf(reqLinePattern, integration))
 	if err != nil {
-		return "", fmt.Errorf("internal error: %v", err)
+		return nil, fmt.Errorf("internal error: %v", err)
 	}
 
-	if groups := exp.FindStringSubmatch(string(output)); groups != nil {
-		return groups[1], nil
+	groups := exp.FindStringSubmatch(lines)
+	if groups == nil {
+		return nil, nil
 	}
 
-	return "", nil
+	version, err := parseVersion(groups[1])
```

Comment:
What if, for some strange reason, multiple lines match the given integration name? Should we handle this?

Reply:
The only reason I see why this could happen, is if we have a duplicated line in our requirement.txt that is read to get the `minAllowedVersion`. And the place where we should handle this is in the CI of integrations-core to make sure that doesn't happen IMO.

Reply:
Sure, I understand, I just think we could add a few lines of defensive programming :) I will defer to @olivielpeau here.